### PR TITLE
doc: esp32: update flashing instructions

### LINKED ISSUE #9389: ESP32 support: setting env var ESP_DEVICE not working
refer to doc
http://docs.zephyrproject.org/boards/xtensa/esp32/doc/esp32.html

Quoted as:
"Environment variables can be set to set the serial port device, baud rate, and other options. Please refer to the following table for details."

after setting ESP_DEVICE=/dev/ttyUSB1,  'ninja flash' still flashing to port ttyUSB0.

Since I can not find ESP_DEVICE env setting in code, I am wondering if the doc is out-of-date or I used it wrongly.

thanks.

### PR DESCRIPTION
**Pull request by nashif**
Details about using west for flashing instead of relying on environment
variables.

Fix #9389.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

**Comment by ceolin** [comment]
Lgtm :+1:

**Comment by nashif** [comment]
please +1 it then

File: boards/xtensa/esp32/doc/esp32.rst [hunk 1]
```diff
@@ -132,10 +112,37 @@ etc).  It might be necessary to change the flash frequency or the flash
 mode; please refer to the `esptool documentation`_ for guidance on these
 settings.
 
-If ``ESP_TOOL`` is set to "espidf", the `esptool.py`_ script found within
-ESP-IDF will be used.  Otherwise, this variable is handled as a path to
-the tool.
+All flashing options are now handled by the ``west`` tool, to flash with custom
```
Comment (dbkinder) [inline boards/xtensa/esp32/doc/esp32.rst]: How about:

    All flashing options are now handled by the :ref:`west` tool, including flashing
    with custom options such as a different serial port.  The ``west`` tool supports
    specific options for the ESP32 board, as listed here:

I notice that there's no mention of these ESP32-specific options in the main west documentation (doc/west/index.rst).  That doesn't seem right?

**Review by dbkinder — DISMISSED** [review]

File: boards/xtensa/esp32/doc/esp32.rst (hunk 1 above)
Comment (nashif) [reply boards/xtensa/esp32/doc/esp32.rst]: > I notice that there's no mention of these ESP32-specific options in the main west documentation (doc/west/index.rst). That doesn't seem right?

yeah, that is expected.

**Pull request merged**